### DISCUSSION TITLE
Getting ip, port, protocol info for mail, caldav, ldap

### DISCUSSION THREAD

**Discussion by ghost** [ecederstrand/exchangelib#1299] [Q&A]
Can we discover the connection addresses and ports of these protocols using exchangelib?
POP/IMAP/SMTP/Caldav/Carddav/LDAP

**Comment by ecederstrand** [comment]
Some of this may be available in the autodiscover information available through `Account.ad_response` if you conected via autodiscover. Otherwise, I don't think EWS provides this information.

**Reply by ghost** [reply]
```Account.ad_response``` it gaves me only a url

**Comment by ghost** [comment]
Workaround solution, that it can be done with EWS:[davmail](https://github.com/mguessan/davmail), it is for creating gateway. I don't need gateway, i need proper address:port for these protocols. But if we can create gateway with exchangelib it also works for me. another solution [autodiscover-email-settings](https://github.com/Monogramm/autodiscover-email-settings)
